medicines/web/src/pages/index.tsx

```diff
@@ -15,31 +15,45 @@ const App: React.FC = () => {
 
   const router = useRouter();
   const {
-    query: { search, page, substance, disclaimer, doc },
+    query: {
+      search: searchQS,
+      page: pageQS,
+      substance: substanceQS,
+      disclaimer: disclaimerQS,
+      doc: docQS,
+      product: productQS,
+    },
   } = router;
 
   useEffect(() => {
     Events.viewPage('homepage');
   }, []);
 
   useEffect(() => {
-    if (search) {
-      const pageNum = page || 1;
+    // handle and redirect legacy urls
+    if (searchQS) {
+      const page = pageQS || 1;
+      const pathname = productQS ? '/product' : '/search';
```

Comment:
just to confirm my understanding: this is here to reroute from the old query strings to the new pages?